ts/nni_manager/core/tuner_command_channel/websocket_channel.ts

```diff
@@ -142,6 +142,7 @@ class WebSocketChannelImpl implements WebSocketChannel {
         }
 
         this.serving = false;
+        this.waitingPong = false;
```

Comment:
what's this used for?

Reply:
There is a callback at 5 seconds intervals to handle ping.
It send a ping to client and set `waitingPong` to `true`.
In 5 seconds, if the client has send back a pong, the pong handler will set `waitingPong` to `false`.
5 seconds later, the next callback checks `waitingPong`'s value and if it's `true`, the client is considered lost responsive.

This line is executed at establishing of a new connection and effectively skips the first pong check.
This is because in UT the pong interval is reduced to a short 10ms; therefore the client might not be able to finish all on connect jobs and send first pong in time.
In real world this also somehow increases robustness. The drawback is if the Python client does fail on reconnection init, it will take an extra 5s to detect.